### PR TITLE
Fix event handler run order and created tests to verify the fix

### PR DESCRIPTION
**Pull request by mateuscb**
**What**: 
- Fixes the order the handler is called in respect to the middleware. The handler should be called as a `next` and not after all the middleware have been called. Inspired by: https://github.com/ambelovsky/koa-socket-2/commit/fb0192bfed2f8d2cf0929b9a89a0f2644161d3c3#diff-9b2f90c89d460ee80ced18e01748824eR44
- Added a test to verify the fix worked as expected

**Why**: Fix the issue

**How**: Fixed the bug

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged 
- [x] Added myself to contributors table

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -1,3 +1,23 @@
+/*eslint-disable */
```
Comment (mateuscb) [inline jest.config.js]: Couldn't figure this one out. Something kept adding the `;` to lines 1 and 11, but then lint would complain saying there were `;`. Would be nice to figure this one out and remove the `eslint-disable`. Could use your help here.

File: src/socket.ts [hunk 2]
```diff
@@ -51,14 +51,18 @@ export default class Socket {
...
-        return;
-      }
+      this.invokeHandler(handler, ctx, message);
+    });
+  }
 
-      this.middleware(ctx).then(() => {
-        handler(ctx, message);
-      });
+  invokeHandler(handler: EventHandler, ctx: SocketContext, message: any) {
```
Comment (mateuscb) [inline src/socket.ts]: I split this out so it would be easier to test. Thoughts?

**Comment by pbomb** [comment]
@mateuscb the TravisCI build is failing because your installed dependencies are pointing to your company's local NPM registry. You can fix this by using a different `~/.npmrc` file.

Additionally, please add yourself to the list of contributors by running `yarn run add-contributor`.

**Comment by pbomb** [comment]
Oops, sorry, I see now that you did add yourself as a contributor. Thanks!

**Pull request merged**